simulator/risc_v/riscv_instr.cpp

```diff
@@ -13,103 +13,110 @@
-template<typename I> const auto execute_blt = ALU::branch<I, ALU::lt<I>>;
-template<typename I> const auto execute_bge = ALU::branch<I, ALU::ge<I>>;
-template<typename I> const auto execute_bltu = ALU::branch<I, ALU::ltu<I>>;
-template<typename I> const auto execute_bgeu = ALU::branch<I, ALU::geu<I>>;
-template<typename I> const auto execute_load = ALU::addr<I>;
-template<typename I> const auto execute_store = ALU::store_addr<I>;
-template<typename I> const auto execute_addi = ALU::riscv_addition_imm<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_addiw = ALU::riscv_addition_imm<I, uint32>;
-template<typename I> const auto execute_addid = ALU::riscv_addition_imm<I, uint64>;
-template<typename I> const auto execute_slti = ALU::set<I, ALU::lti<I>>;
-template<typename I> const auto execute_sltiu = ALU::set<I, ALU::ltiu<I>>;
-template<typename I> const auto execute_xori = ALU::xori<I>;
-template<typename I> const auto execute_ori = ALU::ori<I>;
-template<typename I> const auto execute_andi = ALU::andi<I>;
-template<typename I> const auto execute_slli = ALU::sll<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_slliw = ALU::sll<I, uint32>;
-template<typename I> const auto execute_sllid = ALU::sll<I, uint64>;
-template<typename I> const auto execute_srli = ALU::srl<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_srliw = ALU::srl<I, uint32>;
-template<typename I> const auto execute_srlid = ALU::srl<I, uint64>;
-template<typename I> const auto execute_srai = ALU::sra<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_sraiw = ALU::sra<I, uint32>;
-template<typename I> const auto execute_sraid = ALU::sra<I, uint64>;
-template<typename I> const auto execute_add = ALU::riscv_addition<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_addw = ALU::riscv_addition<I, uint32>;
-template<typename I> const auto execute_sub = ALU::riscv_subtraction<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_subw = ALU::riscv_subtraction<I, uint32>;
-template<typename I> const auto execute_sll = ALU::sllv<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_sllw = ALU::sllv<I, uint32>;
-template<typename I> const auto execute_slt = ALU::set<I, ALU::lt<I>>;
-template<typename I> const auto execute_sltu = ALU::set<I, ALU::ltu<I>>;
-template<typename I> const auto execute_xor = ALU::xorv<I>;
-template<typename I> const auto execute_srl = ALU::srlv<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_srlw = ALU::srlv<I, uint32>;
-template<typename I> const auto execute_sra = ALU::srav<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_sraw = ALU::srav<I, uint32>;
-template<typename I> const auto execute_or = ALU::orv<I>;
-template<typename I> const auto execute_and = ALU::andv<I>;
+template<typename I> const auto execute_lui    = RISCVALU<I>::template upper_immediate<12>;
```

Comment:
Please put indentation back. We are not going to change whole table if a long-name-instruction is added.

Reply:
We already have such indentation in mips_instr.cpp. Still return?

Reply:
Yes. MIPS is mostly finalized, while RISC-V is being extended.